query/src/frontend/influxrpc.rs

```diff
@@ -1459,8 +1504,113 @@ fn make_selector_expr(
         .alias(column_name))
 }
 
+/// Creates specialized / normalized predicates
+#[derive(Debug)]
+struct PredicateNormalizer {
+    unnormalized: Predicate,
+    normalized: HashMap<String, TableNormalizedPredicate>,
+}
+
+impl PredicateNormalizer {
+    fn new(unnormalized: Predicate) -> Self {
+        Self {
+            unnormalized,
+            normalized: Default::default(),
+        }
+    }
+
+    /// Return a reference to the unnormalized predicate
+    fn unnormalized(&self) -> &Predicate {
+        &self.unnormalized
+    }
+
+    /// Return a reference to a predicate specialized for `table_name`
+    fn normalized(&mut self, table_name: &str) -> Arc<Predicate> {
+        if let Some(normalized_predicate) = self.normalized.get(table_name) {
+            return normalized_predicate.inner();
+        }
+
+        let normalized_predicate =
+            TableNormalizedPredicate::new(table_name, self.unnormalized.clone());
+
+        self.normalized
+            .entry(table_name.to_string())
+            .or_insert(normalized_predicate)
+            .inner()
+    }
+}
+
+/// Predicate that has been "specialized" / normalized for a
+/// particular table. Specifically:
+///
+/// * all references to the [MEASUREMENT_COLUMN_NAME] column in any
+/// `Exprs` are rewritten with the actual table name
```

Comment:
```suggestion
/// * all references to the [MEASUREMENT_COLUMN_NAME] column in any
/// `Exprs` are rewritten with the actual table name
///
/// For example if the original predicate was 
/// ```text
/// _measurement = "some_table"
/// ```
/// 
/// When evaluated on table "cpu" then the predicate is rewritten to 
/// ```text
/// "cpu" = "some_table"
///
```